### PR TITLE
[HB-6679] Partner Consents

### PR DESCRIPTION
**Pull request by bichenwang**

### CODE REVIEW

**Comment by bichenwang** [comment]
this isn't going to compile until https://github.com/ChartBoost/chartboost-core-android-sdk-private/pull/102 gets merged and I make a temporary 0.4.0 on private maven then retrigger.

File: UsercentricsAdapter/src/main/java/com/chartboost/core/consent/usercentrics/UsercentricsAdapter.kt [hunk 1]
```diff
@@ -438,6 +466,33 @@ class UsercentricsAdapter() : ConsentAdapter, InitializableModule {
         }
     }
 
+    private fun updatePartnerConsents(
+        consents: List<UsercentricsServiceConsent>,
+        notify: NotificationType,
+        oldPartnerConsents: Map<String, ConsentStatus>
```
Comment (chauduyphanvu) [inline UsercentricsAdapter/src/main/java/com/chartboost/core/consent/usercentrics/UsercentricsAdapter.kt:472]:
```suggestion
        oldPartnerConsents: Map<String, ConsentStatus>,
```

File: UsercentricsAdapter/src/main/java/com/chartboost/core/consent/usercentrics/UsercentricsAdapter.kt [hunk 2]
```diff
@@ -438,6 +466,33 @@ class UsercentricsAdapter() : ConsentAdapter, InitializableModule {
...
+            when (notify) {
+                NotificationType.DIFFERENT_FROM_CURRENT_VALUE -> if (previousConsentStatus != newConsentStatus) Utils.safeExecute {
+                    listener?.onPartnerConsentStatusChange(partnerId, newConsentStatus)
+                }
+
+                NotificationType.DIFFERENT_FROM_CACHED_VALUE -> if ((oldPartnerConsents[partnerId] ?: ConsentStatus.UNKNOWN) != newConsentStatus) Utils.safeExecute {
+                    listener?.onPartnerConsentStatusChange(partnerId, newConsentStatus)
+                }
+
+                else -> Unit
```
Comment (cb-jpadilla) [inline UsercentricsAdapter/src/main/java/com/chartboost/core/consent/usercentrics/UsercentricsAdapter.kt:488]: Is this similar to just saying `{}`

**Review by cb-jpadilla — APPROVED** [review]
✔️

File: UsercentricsAdapter/src/main/java/com/chartboost/core/consent/usercentrics/UsercentricsAdapter.kt (hunk 2 above)
Comment (bichenwang) [reply UsercentricsAdapter/src/main/java/com/chartboost/core/consent/usercentrics/UsercentricsAdapter.kt:488]: Yep. Means do nothing

**Pull request merged**